### PR TITLE
add setup-r-dependencies

### PR DESCRIPTION
**Pull request by vedhav**
Related to https://github.com/insightsengineering/nestdevs-tasks/issues/65
Switch to setup-r-dependencies.
Added `teal.reporter` for lookup-refs.

### CODE REVIEW

**Review by m7pr — APPROVED** [review]

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -41,6 +41,12 @@ jobs:
         checking top-level files .* NOTE
       unit-test-report-brand: >-
         https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/thumbs/teal.png
+      deps-installation-method: setup-r-dependencies
+      lookup-refs: |
+        insightsengineering/rlistings
```
[the remark this answers is not in the document]
Comment (vedhav) [reply .github/workflows/check.yaml:46]: I think it's alright. The installation would have failed if we did not include something important. From what I observed adding the packages in the DESCRIPTION file was sufficient, except for `teal.slice` and `formatters`

[the remark this answers is not in the document]
Comment (vedhav) [reply .github/workflows/check.yaml:46]: And we can see that the installation was successful. Without any one of these the CI will fail.

[the remark this answers is not in the document]
Comment (vedhav) [reply .github/workflows/check.yaml:46]: Which is not the case for rlistings

[the remark this answers is not in the document]
Comment (m7pr) [reply .github/workflows/check.yaml:46]: ok

**Pull request merged**